mayastor/src/bdev/storage.rs

```diff
@@ -0,0 +1,69 @@
+//! Definition of StorageBdev trait that prescribes how to create
+//! and destroy bdevs from an appropriate device URI.
+//! Note also that the create() and destroy() methods therein
+//! are declared to be async.
+//!
+//! Support for any new device type that we may wish to add effectively
+//! amounts to providing an appropriate implementation of this trait
+//! (as well as a suitable implementation of the TryFrom<&Url> trait
+//! in order to be able to parse the corresponding URI).
+//!
+//! Creating a bdev for any supported device type should now be as simple as:
+//! ```ignore
+//!     let uri = "aio:///tmp/disk1.img?blk_size=512";
+//!     StorageUri::parse(&uri)?.create().await?;
+//! ```
+
+use std::convert::TryFrom;
+
+use async_trait::async_trait;
+use snafu::ResultExt;
+use url::Url;
+
+use crate::nexus_uri::{self, BdevCreateDestroy};
+
+mod aio;
+mod bdev;
+mod iscsi;
+mod nvmf;
+mod uring;
+
+#[async_trait]
+/// Definition of trait that any supported
+/// device type is required to implement.
+pub trait StorageBdev {
+    type Error;
+    fn get_name(&self) -> String;
```

Comment:
This should not be part of the trait,  what would it? Who cares how you destroy the bdev right?